packages/react/src/components/ImageComponent/ImageComponent.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import PropTypes from 'prop-types';
+import React from 'react';
+import { settings as ddsSettings } from '@carbon/ibmdotcom-utilities';
+import { settings } from 'carbon-components';
+
+const { stablePrefix } = ddsSettings;
+const { prefix } = settings;
+
+/**
+ *  sorts media query min-widths order to ensure the
+ * browser returns the proper sources and the specified widths
+ *
+ * @param {Array} sources image sources and min-widths
+ *
+ * @returns {Array} sorted array of sources
+ */
+const sortSources = sources => {
+  return sources.sort((a, b) => (a.minWidth > b.minWidth ? -1 : 1));
+};
+
+/**
+ * renders background image
+ *
+ * @param {object} props props object
+ * @param {object} props.images array of images used for diff breakpoints
+ * @param {string} props.defaultImage default image (usually image for largest breakpoint)
+ * @param {string} props.alt alt of the image
+ * @returns {*} picture element
+ */
+const Image = ({ images, defaultImage, alt }) => {
+  const sortedImages = sortSources(images);
+
+  return (
+    <picture data-autoid={`${stablePrefix}--imagecomponent`}>
+      {sortedImages.map((imgSrc, key) => {
+        return <source key={key} srcSet={imgSrc.url} />;
+      })}
+      <img
+        className={`${prefix}--imagecomponent__image`}
+        src={defaultImage}
+        alt={alt}
+      />
+    </picture>
+  );
+};
+
+Image.propTypes = {
+  images: PropTypes.arrayOf(
+    PropTypes.shape({
+      minWidth: PropTypes.number,
```

Comment:
I'm wondering if we should be referencing the breakpoint names rather than actual numbers here (e.g. `sm`, `md`, `lg` `xlg`, 'max`), then we would render out the mappings rather than have the user rely on getting the numbers right?

https://www.carbondesignsystem.com/guidelines/2x-grid/implementation#responsive-options

Reply:
We should probably allow for both. There are moments where the adopters might need to break something down in between the standard breakpoints. So something like a `switch` or `if` statement that allows us to catch the keywords first and set the appropriate breakpoint, but then on `default` set the breakpoint to whatever the user passed in.

Reply:
I added this as a comment over in #970, and thought it might be useful in this scenario for you to consider as well. Carbon exports their `breakpoints` in Javascript allowing us to use them for things like this.

https://github.com/carbon-design-system/carbon/blob/master/packages/layout/src/index.js#L45-L71

Reply:
updated @photodow